### PR TITLE
storage: move some wipe logic out to functions

### PR DESCRIPTION
**Pull request by pro-wh**
run the deferred `rows.Close()` earlier

### CODE REVIEW

**Review by mitjat — APPROVED** [review]
Yeah, tasty. Thanks.

Linter is complaining about a `err` value being lost somewhere, but it looks like the line number is missing? :confused:

File: storage/postgres/client.go [hunk 1]
```diff
@@ -237,21 +237,7 @@ func (c *Client) listIndexerTables(ctx context.Context) ([]string, error) {
...
-	for _, table := range tables {
-		c.logger.Info("dropping table", "table", table)
-		if _, err = c.pool.Exec(ctx, fmt.Sprintf("DROP TABLE %s CASCADE;", table)); err != nil {
-			return err
-		}
-	}
-
-	// List, then drop all custom types.
-	// Query from https://stackoverflow.com/questions/3660787/how-to-list-custom-types-using-postgres-information-schema
+func (c *Client) listIndexerTypes(ctx context.Context) ([]string, error) {
```
Comment (mitjat) [inline storage/postgres/client.go:240]: That's `listNexusTypes` to you, mister :roll_eyes: 
https://github.com/oasisprotocol/nexus/pull/467

Comment (pro-wh) [reply storage/postgres/client.go]: renamed! 🎨

**Comment by pro-wh** [comment]
ah yeah nice catch lint

**Pull request merged**